src/java.base/share/classes/java/lang/System.java

```diff
@@ -2113,6 +2117,12 @@ private static void initPhase1() {
         VM.saveProperties(tempProps);
         props = createProperties(tempProps);
 
+        // Check if sun.jnu.encoding is supporetd or not
+        var jnuEncoding = props.getProperty("sun.jnu.encoding");
+        if (jnuEncoding == null || !Charset.isSupported(jnuEncoding)) {
+            notSupportedJnuEncoding = jnuEncoding == null ? "null" : jnuEncoding;
```

Comment:
I think we are nearly there. When setting notSupportedJnuEncoding then we also need to set the sun.jnu.encoding property to "UTF-8", setting it in initPhase3 is too late because the file system may be used using initPhase2.

Reply:
Initially, I thought that was the case, but I mistook your suggestion as setting only `notSupportedJnuEncoding` in initPhase1. Reverted the fallback location into `initPhase1()`.

Reply:
> Initially, I thought that was the case, but I mistook your suggestion as setting only `notSupportedJnuEncoding` in initPhase1. Reverted the fallback location into `initPhase1()`.

np, it's always tricky to touch code that executes early in the startup. I think we are good now.